src/variable.cc

```diff
@@ -38,6 +38,9 @@ Variable &Variable::operator=(const Variable &other) {
           ? new std::unordered_map<variable_value_t, variable_value_index_t>(
                 *other.domain_map)
           : nullptr);
+  domain_list.reset(other.domain_list
+                        ? new std::vector<variable_value_t>(*other.domain_list)
+                        : nullptr);
```

Comment:
I think this is unnecessary as domain_list is reconstructed from the map when nullptr but fine to keep for consistency and to minimize surprise